src/indexApi.ts

```diff
@@ -1,18 +1,108 @@
+  status: asString,
+  value: asObject({
+    _id: asString,
+    data: asObject({ minAmount: asString })
+  })
+})
 
 // call the packages we need
 const app = express()
 
+// Nano for CouchDB
+// =============================================================================
+const nanoDb = nano(config.dbFullpath)
+const swapMinAmts = mockPlugins.map(plugin => {
+  const pluginMinAmt = nanoDb.db.use(minAmtDbPrefix + plugin.pluginName)
+  return promisify(pluginMinAmt)
+})
+
+// ROUTES FOR OUR API
+// =============================================================================
+const router = express.Router()
+
+router.use(function (req, res, next) {
+  console.log('Something is happening.')
+  next() // make sure we go to the next routes and don't stop here
+})
+
+router.get('/getMinimum/', async function (req, res) {
+  const currencyPair: string = req.query.currencyPair // Param of 'currencyPair' in URI
+  const currentUTCDate: string = formatISO(new Date(), {
+    representation: 'date'
+  }) // Variable for current UTC date using date-fns
+  try {
+    // Array of promises to get the latest documents for the min amount data per plugin
+    const docPromisesArr = swapMinAmts.map(async pluginDb => {
+      return pluginDb.get(currentUTCDate + ':' + currencyPair)
+    })
+    // Capture the result of each promise, including whether it was fulfilled or rejected
+    const docs = await Promise.allSettled(docPromisesArr)
+
+    // Use 'reduce' to go through settled promises and find the lowest minAmount amongst the plugins
+    const minAmountFloat = docs.reduce((currentMinAmount, doc) => {
+      try {
+        const cleanedDoc = asFulfilledPromise(doc) // Clean doc for a fulfilled promise
+        const pluginMinAmount = +cleanedDoc.value.data.minAmount // Grab the plugin's minAmount for the currencyPair
+        // If the plugin's minAmount is lower than currentMinAmount, update currentMinAmount
+        return pluginMinAmount < currentMinAmount
+          ? pluginMinAmount
+          : currentMinAmount
+      } catch {
+        // Promise was rejected or fulfilled promise returned invalid data
+        return currentMinAmount
+      }
+    }, Infinity) // Start currentMinAmount at Infinity
+
+    // Check if minAmountFloat is still Infinity
+    if (!isFinite(minAmountFloat)) {
+      // No data could be found in the database, throw error
+      throw new Error('not_found')
+    }
+
+    res.json({
+      currencyPair,
+      date: currentUTCDate,
+      minAmount: minAmountFloat.toString()
+    })
+  } catch (e) {
+    console.log(e)
+    let errorCode, errorType, errorMsg
+    if (e != null && e.message === 'not_found') {
+      errorCode = 404
+      errorType = 'not_found'
+      errorMsg = `SWAP SERVER: All lookups failed to find minimum amount for currencyPair ${currencyPair} at date ${currentUTCDate}.`
+    } else {
+      errorCode = 500
+      errorType = 'internal_server'
+      errorMsg = 'Internal Server Error.'
+    }
```

Comment:
You should have predefined constant Error types that can be expended with an optional message.

Reply:
Created a util for error responses.